### PR TITLE
(feat) Migrating programs form to use RHF and Zod

### PR DESCRIPTION
**Pull request by lumuchris256**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Migrated the form to use RHF and Zod

## Screenshots
[Screencast from 2023-07-04 16-57-48.webm](https://github.com/openmrs/openmrs-esm-patient-chart/assets/59338693/a054eae6-e68e-4d55-a065-331f827d3ebb)

## Related Issue
https://issues.openmrs.org/browse/O3-2140

## Other
cc @denniskigen @hadijahkyampeire

### CODE REVIEW

File: packages/esm-patient-programs-app/src/programs/programs-form.component.tsx [hunk 1]
```diff
@@ -146,152 +160,149 @@ const ProgramsForm: React.FC<ProgramsFormProps> = ({ closeWorkspace, patientUuid
...
-      currentEnrollment,
-      mutateEnrollments,
-      closeWorkspace,
-      t,
-    ],
+    [patientUuid, currentEnrollment, mutateEnrollments, closeWorkspace, t],
   );
+
+  const onError = (error) => {
+    console.error(error);
```
Comment (hadijahkyampeire) [inline packages/esm-patient-programs-app/src/programs/programs-form.component.tsx:167]: why are you console logging the error, can we do something else with this function or just get rid of it if we don't need it.

File: packages/esm-patient-programs-app/src/programs/programs-form.scss [hunk 2]
```diff
@@ -33,3 +34,9 @@
   justify-content: space-between;
   height: calc(100vh - 6rem);
 }
+
+.errorMessage {
+  @include type.type-style("label-02");
+  margin-top: 0.5rem;
```
Comment (hadijahkyampeire) [inline packages/esm-patient-programs-app/src/programs/programs-form.scss:40]:
```suggestion
  margin-top: 0.5rem;
  margin-top: spacing.$spacing-03;
```
Remember to import the spacing 
@use "@carbon/styles/scss/spacing";

**Review by hadijahkyampeire — APPROVED** [review]
Mostly looks great, just a few minor suggestions. Thanks @lumuchris256

File: packages/esm-patient-programs-app/src/programs/programs-form.component.tsx (hunk 1 above)
Comment (lumuchris256) [reply packages/esm-patient-programs-app/src/programs/programs-form.component.tsx]: reslolved

**Pull request merged**